extensions/blocks/donations/view.js

```diff
@@ -20,226 +20,225 @@ import { initializeMembershipButtons } from '../../shared/memberships';
+	updateAmountFromCustomAmountInput() {
+		const input = this.block.querySelector( '.donations__custom-amount .donations__amount-value' );
+		const wrapper = this.block.querySelector( '.donations__custom-amount' );
 
-	// Make input editable.
-	input.setAttribute( 'contenteditable', '' );
+		const amount = input.innerHTML;
+		if ( ! amount ) {
+			this.amount = null;
+			this.toggleDonateButton( false );
+			return;
+		}
 
-	// Prevent new lines.
-	input.addEventListener( 'keydown', event => {
-		if ( event.keyCode === 13 ) {
-			event.preventDefault();
+		// Validates the amount.
+		const currency = input.dataset.currency;
+		const parsedAmount = parseAmount( amount, currency );
+		if ( parsedAmount && parsedAmount >= minimumTransactionAmountForCurrency( currency ) ) {
+			wrapper.classList.remove( 'has-error' );
+			this.amount = parsedAmount;
+			this.toggleDonateButton( true );
+		} else {
+			wrapper.classList.add( 'has-error' );
+			this.amount = null;
+			this.toggleDonateButton( false );
 		}
-	} );
+		this.updateUrl();
+	}
 
-	input.addEventListener( 'focus', () => {
-		// Add focus styles to wrapper element.
-		wrapper.classList.add( 'has-focus' );
-		wrapper.classList.remove( 'is-selected' );
+	initNavigation() {
+		const navItems = this.block.querySelectorAll( '.donations__nav-item' );
+		const tabContent = this.block.querySelector( '.donations__tab' );
+		const tabContentClasses = {
+			'one-time': 'is-one-time',
+			'1 month': 'is-monthly',
+			'1 year': 'is-annual',
+		};
+
+		const handleClick = event => {
+			// Update selected interval.
+			const prevInterval = this.interval;
+			const newInterval = event.target.dataset.interval;
+			this.interval = newInterval;
+
+			// Toggle nav item.
+			const prevNavItem = this.getNavItem( prevInterval );
+			prevNavItem.classList.remove( 'is-active' );
+			const newNavItem = this.getNavItem( newInterval );
+			newNavItem.classList.add( 'is-active' );
+
+			// Toggle tab content.
+			tabContent.classList.remove( tabContentClasses[ prevInterval ] );
+			tabContent.classList.add( tabContentClasses[ newInterval ] );
+
+			// Reset chosen amount.
+			this.amount = null;
+			this.isCustomAmount = false;
+			this.resetSelectedAmount();
+			this.updateUrl();
+
+			// Disable donate button.
+			this.toggleDonateButton( false );
+		};
+
+		navItems.forEach( navItem => {
+			navItem.addEventListener( 'click', handleClick );
+			navItem.addEventListener( 'keydown', handleClick );
+		} );
+
+		// Activates the default tab on first execution.
+		const navItem = this.getNavItem( this.interval );
+		navItem.classList.add( 'is-active' );
+		tabContent.classList.add( tabContentClasses[ this.interval ] );
+	}
 
-		// Toggle selected amount.
-		resetSelectedAmount();
-		if ( jetpackDonationsIsCustomAmount ) {
+	handleCustomAmount() {
+		const input = this.block.querySelector( '.donations__custom-amount .donations__amount-value' );
+		if ( ! input ) {
```

Comment:
Nice early return! This could also be one line, but that doesn't seem like the code style